modules/shared/src/main/scala/org/tessellation/currency/schema/EstimatedFee.scala

```diff
@@ -0,0 +1,15 @@
+package org.tessellation.currency.schema
+
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.balance.Amount
+
+import derevo.cats.{order, show}
+import derevo.circe.magnolia.encoder
+import derevo.derive
+
+@derive(encoder, order, show)
+case class EstimatedFee(fee: Amount, address: Option[Address])
```

Comment:
I don't understand the purpose of the `Option[Address]` in the fee estimation. For the Fee itself I can see needing an `Address` and `Amount` but the use here is unclear to me.

Reply:
@sdavidp If you say so, it should be either no fee or fee + address. Right now you allow an impossible state to be defined: fee + no address.

So it should be defined as below:
```
sealed trait EstimatedFee

case object Zero extends EstimatedFee
case class Estimated(amount: Amount, address: Address) extends EstimatedFee

```

Reply:
@marcinwadon Ok, I will change the implementation. However, the ticket wants the encoded JSON for _zero_ to be `{ "fee" : 0 }`

Reply:
@marcinwadon Updated.